pkg/webhook/default_server/webhook_test.go

```diff
@@ -0,0 +1,184 @@
+package defaultserver
+
+import (
+	"bytes"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"io"
+	"net/http"
+	"net/http/httputil"
+	"os"
+
+	"crypto/tls"
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/onsi/gomega"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var c client.Client
+
+const timeout = time.Second * 10
+const webhookURL = "https://localhost:9876/mutating-create-function"
+
+// Integration test for webhook
+// Spin up the webhook server and issue an admission request against it with an invalid function
+func TestWebHook(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	g.Expect(createCertificates(t)).NotTo(gomega.HaveOccurred())
+
+	// create manager
+	mgr, err := manager.New(cfg, manager.Options{})
+	g.Expect(err).NotTo(gomega.HaveOccurred())
+	// add webhook to manager
+	g.Expect(Add(mgr)).NotTo(gomega.HaveOccurred())
+
+	// start manager
+	stopMgr, mgrStopped := StartTestManager(mgr, g)
+	defer func() {
+		close(stopMgr)
+		mgrStopped.Wait()
+	}()
+
+	// create admission request
+	var admissionReview = getAdmissionRequest()
+	jsonRequest, err := json.Marshal(admissionReview)
+	if err != nil {
+		t.Errorf("Error encoding admission review: %v", err)
+	}
+
+	// wait for webserver to be reachable
+	g.Eventually(func() error {
+		_, err := getInsecureClient().Post(webhookURL, "application/json", bytes.NewBuffer(jsonRequest))
+		return err
+	}, timeout).Should(gomega.Succeed())
+
+	// get admission response from webhook
+	admissionRequest, err := getInsecureClient().Post(webhookURL, "application/json", bytes.NewBuffer(jsonRequest))
+	if err != nil {
+		t.Errorf("Could not get result from webhook: %v", err)
+	}
+	defer admissionRequest.Body.Close()
+	g.Expect(admissionRequest.StatusCode).To(gomega.Equal(200))
+
+	// print admission request
+	byts, _ := httputil.DumpResponse(admissionRequest, true)
+	t.Logf("response: %v", string(byts))
+
+	// decode admission review
+	var response admissionv1beta1.AdmissionReview
+	bodyBytes, err := ioutil.ReadAll(admissionRequest.Body)
+	if err != nil {
+		t.Error(err)
+	}
+	json.Unmarshal(bodyBytes, &response)
```

Comment:
this can be a single line:  `err = json.NewDecoder(admissionRequest.Body).Decode(&response)`

Reply:
Thanks :)